pkg/engine/resource/object_key.go

```diff
@@ -67,6 +72,30 @@ func isNamespaced(gvk schema.GroupVersionKind, di discovery.CachedDiscoveryInter
 	return false, fmt.Errorf("a resource with GVK %v seems to be missing in API resource list", gvk)
 }
 
+func isKnownType(gvk schema.GroupVersionKind, di discovery.CachedDiscoveryInterface) (bool, error) {
+	// First try, this may return nil because of the cache
+	apiResource, err := getAPIResource(gvk, di)
+	if err != nil {
+		return false, err
+	}
+	if apiResource != nil {
+		return true, nil
+	}
+
+	// Second try, now with invalidated cache. If we still get nil, we know it's not there.
+	log.Printf("Failed to get APIResource for %v, retry with invalidated cache.", gvk)
+	di.Invalidate()
+	apiResource, err = getAPIResource(gvk, di)
```

Comment:
Wdyt about making these next few lines a nested `isKnown` method? The same goes for the above `isNamespaced` method.

Reply:
Not sure about the inline func, I tried it and it didn't really help much. But you're right that there's a bit of duplication, so I extracted a `getUncachedAPIResource`